cg/services/analysis_starter/configurator/implementations/balsamic.py

```diff
@@ -0,0 +1,306 @@
+            account=self.slurm_account,
+            binary=self.balsamic_binary,
+            conda_binary=self.conda_binary,
+            cluster_config=self.default_cluster_config,
+            environment=self.balsamic_binary,
+            mail_user=self.slurm_mail_user,
+            qos=self.store.get_case_by_internal_id(case_id).slurm_priority,
+            sample_config=self._get_sample_config_path(case_id),
+        )
+        balsamic_config: BalsamicCaseConfig = self._set_flags(config=balsamic_config, **flags)
+        self._ensure_valid_config(balsamic_config)
+        return balsamic_config
+
+    def _build_cli_input(self, case_id) -> BalsamicConfigInput:
+        case: Case = self.store.get_case_by_internal_id(case_id)
+        if self._all_samples_are_wgs(case):
+            return self._build_wgs_config(case)
+        else:
+            return self._build_targeted_config(case)
+
+    def _build_wgs_config(self, case: Case) -> BalsamicConfigInput:
+        patient_sex: SexOptions = self._get_patient_sex(case)
+        return BalsamicConfigInput(
+            analysis_dir=self.root_dir,
+            analysis_workflow=case.data_analysis,
+            artefact_snv_observations=self.loqusdb_artefact_snv,
+            balsamic_binary=self.balsamic_binary,
+            balsamic_cache=self.cache_dir,
+            cadd_annotations=self.cadd_path,
+            cancer_germline_snv_observations=self.loqusdb_cancer_germline_snv,
+            cancer_germline_sv_observations=self.loqusdb_cancer_germline_sv,
+            cancer_somatic_snv_observations=self.loqusdb_cancer_somatic_snv,
+            cancer_somatic_sv_observations=self.loqusdb_cancer_somatic_sv,
+            case_id=case.internal_id,
+            clinical_snv_observations=self.loqusdb_clinical_snv,
+            clinical_sv_observations=self.loqusdb_clinical_sv,
+            conda_binary=self.conda_binary,
+            fastq_path=Path(self.root_dir, case.internal_id, "fastq"),
+            gender=patient_sex,
+            genome_interval=self.genome_interval_path,
+            genome_version=GenomeVersion.HG19,
+            gens_coverage_pon=(
+                self.gens_coverage_female_path
+                if patient_sex == SexOptions.FEMALE
+                else self.gens_coverage_male_path
+            ),
+            gnomad_min_af5=self.gnomad_af5_path,
+            normal_sample_name=self._get_normal_sample_id(case),
+            sentieon_install_dir=self.sentieon_licence_path,
+            sentieon_license=self.sentieon_licence_server,
+            swegen_snv=self.swen_snv,
+            swegen_sv=self.swen_sv,
+            tumor_sample_name=self._get_tumor_sample_id(case),
+        )
+
+    def _build_targeted_config(self, case, **flags) -> BalsamicConfigInput:
+        bed_file: Path = self._resolve_bed_file(case, **flags)
+        patient_sex: SexOptions = self._get_patient_sex(case)
+        return BalsamicConfigInput(
```

Comment:
Sonarcloud is flagging the second instantiation of this as duplication, which is true since most of the fields are populated the same. I could create two models, one for the constant stuff and one for the parts that differ, and then combine them. However that might get confusing. Any thoughts?

Reply:
Will leave like this for now